### PR TITLE
Feature portal38 woadd 964

### PR DESCRIPTION
**Pull request by JMAbreu-acn**
*** PLEASE DO NOT OPEN PULL REQUESTS VIA GITHUB ***

The moodle.git repository at Github is just a mirror of the official repository. We do not accept pull requests at Github.

See CONTRIBUTING.txt guidelines for how to contribute patches for Moodle. Thank you.

--

### CODE REVIEW

File: blocks/gu_spdetails/block_gu_spdetails.php [hunk 1]
```diff
@@ -302,7 +302,7 @@ public static function return_dates($modname, $activity) {
                                    $activity->overridecutoffdate : $activity->cutoffdate;
                 $date->gradingduedate = (!empty($activity->gradingduedate)) ?
                                         $activity->gradingduedate :
-                                        (!(empty($date->duedate)) ? $date->duedate + (86400 * 14) : '0');
+                                        (!(empty($date->duedate)) ? $date->duedate : '0');
```
Comment (francolouietm) [inline blocks/gu_spdetails/block_gu_spdetails.php]: For assignment, `$date->gradingduedate` should only be based on `$activity->gradingduedate`. If this is empty, just set it to 0.

`$date->gradingduedate = (!empty($activity->gradingduedate)) ? $activity->gradingduedate : '0';`

File: blocks/gu_spdetails/block_gu_spdetails.php [hunk 2]
```diff
@@ -317,12 +317,12 @@ public static function return_dates($modname, $activity) {
...
                 $date->gradingduedate = (!empty($activity->cutoffdate)) ?
                                         $activity->cutoffdate :
-                                        (!(empty($date->duedate)) ? $date->duedate + (86400 * 14) : '0');
+                                        (!(empty($date->duedate)) ? $date->duedate : '0');
                 break;
             case 'workshop':
                 $date->gradingduedate = (!empty($activity->gradingduedate)) ?
                                         $activity->gradingduedate :
-                                        (!(empty($date->duedate)) ? $date->duedate + (86400 * 14) : '0');
+                                        (!(empty($date->duedate)) ? $date->duedate : '0');
```
Comment (francolouietm) [inline blocks/gu_spdetails/block_gu_spdetails.php]: Similar to assignment, the `$date->gradingduedate` in workshop should only be based on `$activity->gradingduedate`. If this is empty, just set it to 0.

`$date->gradingduedate = (!empty($activity->gradingduedate)) ? $activity->gradingduedate : '0';`

File: blocks/gu_spdetails/test/block_gu_spdetails_test.php [hunk 3]
```diff
@@ -558,7 +564,7 @@ public function test_return_dates(){
         $expectedAssign2->isdueextended = true;
         $expectedAssign2->startdate = $activity->startdate;
         $expectedAssign2->cutoffdate = $activity->cutoffdate;
-        $expectedAssign2->gradingduedate = $expectedAssign2->duedate + (86400 * 14);
+        $expectedAssign2->gradingduedate = $expectedAssign2->duedate;
```
Comment (francolouietm) [inline blocks/gu_spdetails/test/block_gu_spdetails_test.php]: Update the corresponding test scenarios based on the requested updates on the code.

**Review by francolouietm — CHANGES_REQUESTED** [review]

File: blocks/gu_spdetails/block_gu_spdetails.php (hunk 1 above)
Comment (JMAbreu-acn) [reply blocks/gu_spdetails/block_gu_spdetails.php]: Fixed at commit [1b50080](https://github.com/uofg-andf/andfmoodle/pull/184/commits/1b50080a3cab7bf33148cba444d1b1d2f7579d88)

File: blocks/gu_spdetails/block_gu_spdetails.php (hunk 2 above)
Comment (JMAbreu-acn) [reply blocks/gu_spdetails/block_gu_spdetails.php]: Fixed at commit [1b50080](https://github.com/uofg-andf/andfmoodle/pull/184/commits/1b50080a3cab7bf33148cba444d1b1d2f7579d88)

File: blocks/gu_spdetails/test/block_gu_spdetails_test.php (hunk 3 above)
Comment (JMAbreu-acn) [reply blocks/gu_spdetails/test/block_gu_spdetails_test.php]: Fixed at commit [1b50080](https://github.com/uofg-andf/andfmoodle/pull/184/commits/1b50080a3cab7bf33148cba444d1b1d2f7579d88)

File: blocks/gu_spdetails/block_gu_spdetails.php [hunk 4]
```diff
@@ -333,8 +329,9 @@ public static function return_gradingduedate($finalgrade, $gradingduedate) {
         $duedateobj = new stdClass();
         $duedateobj->hasgrade = (!empty($finalgrade)) ? true : false;
         $duedateobj->gradetext = (!empty($finalgrade)) ? $finalgrade :
-                                get_string('due', SPDETAILS_STRINGS).
-                                userdate($gradingduedate, get_string('convertdate', SPDETAILS_STRINGS));
+                                 ((time() > $gradingduedate) ? ucwords(get_string('overdue', SPDETAILS_STRINGS)) :
```
Comment (francolouietm) [inline blocks/gu_spdetails/block_gu_spdetails.php]: Needs to consider the scenario for 'To be confirmed'.

**Review by francolouietm — CHANGES_REQUESTED** [review]

File: blocks/gu_spdetails/block_gu_spdetails.php (hunk 4 above)
Comment (JMAbreu-acn) [reply blocks/gu_spdetails/block_gu_spdetails.php]: Fixed at commit [7d7091b](https://github.com/uofg-andf/andfmoodle/pull/184/commits/7d7091baa7b2782798a933e84e30c7374545da56)

**Review by francolouietm — APPROVED** [review]

**Pull request merged**